Update troubleshooting.mdx (#18009)
cloudflare/cloudflare-docs: src/content/docs/hyperdrive/observability/troubleshooting.mdx

| `Failed to acquire a connection from the pool.`        | Hyperdrive timed out while waiting for a connection to your database, or cannot connect at all. | If you are seeing this error intermittently, your Hyperdrive pool is being exhausted because too many connections are being held open for too long by your worker. This can be caused by a myriad of different issues, but long-running queries/transactions are a common offender.            |
| `Server connection attempt failed: connection_refused` | Hyperdrive is unable to create new connections to your origin database.                         | A network firewall or access control list (ACL) is likely rejecting requests from Hyperdrive. Ensure you have allowed connections from the public Internet. Sometimes, this can be caused by your database host provider refusing incoming connections when you go over your connection limit. |

> ### Improving performance

Reviewer: ```suggestion
### Improve performance
```

> Having query traffic written as transactions can limit performance. This is because in the case of a transaction, the connection must be held for the duration of the transaction, which limits connection multiplexing.  If there are multiple queries per transaction, this can be particularly impactful on connection multiplexing.  Where possible, we recommend not wrapping queries in transactions to allow the connections to be shared more aggressively.

Reviewer: ```suggestion
Having query traffic written as transactions can limit performance. This is because in the case of a transaction, the connection must be held for the duration of the transaction, which limits connection multiplexing. If there are multiple queries per transaction, this can be particularly impactful on connection multiplexing. Where possible, we recommend not wrapping queries in transactions to allow the connections to be shared more aggressively.
```

> Having query traffic written as transactions can limit performance.  The reason for this is because in the case of a transaction the connection must be held for the duration of the transaction which limits connection multiplexing.  If there are mulitple queries per transaction this can be particularly impactful on connection multiplexing.  Where possible we recommend not wrapping queries in transactions to allow the connections to be shared more aggressively.

Reviewer: ```suggestion
Having query traffic written as transactions can limit performance. This is because in the case of a transaction, the connection must be held for the duration of the transaction, which limits connection multiplexing.  If there are multiple queries per transaction, this can be particularly impactful on connection multiplexing.  Where possible, we recommend not wrapping queries in transactions to allow the connections to be shared more aggressively.
```
